feat(curriculum): forum leaderboard project (#50241)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/643c565936a10109b3f59326.md

---
id: 643c565936a10109b3f59326
title: Step 26
challengeType: 0
dashedName: step-26
---

# --description--

> You have to call the `showLatestPosts` function inside the `fetchData` function. Pass the `data` as an argument to the `showLatestPosts` function. It means that the `showLatestPosts` function will be called after the `fetchData` function is resolved.

Author: fixed

# --hints--

> You have to call the `showLatestPosts` function inside the `fetchData` function, after the `console.log` statement.

Author: fixed

---
id: 643c565936a10109b3f59326
title: Step 9
challengeType: 0
dashedName: step-9
---

# --description--

> To see your changes take effect as you write your `showLatestPosts()` function, call it at the end of your `fetchData()` function. Pass `data` as the argument.

Reviewer: ```suggestion
As you build out your `showLatestPosts()` function, you'll need to call it to see your changes.

Call the `showLatestPosts()` function at the end of your `try` block and pass in `data` for the argument.
```

Author: added

# --hints--

You should call `showLatestPosts()` at the end of your `try` block inside the `fetchData` function.

```js

> assert.match(code, /const\s*fetchData\s*=\s*async\s*\(\)\s*=>\s*{\s*try{\s*const\s*res\s*=\s*await\s*fetch\(FORUM_LATEST\);?\s*const\s*data\s*=\s*await\s*res\.json\(\);?\s*console\.log\(data\);?\s*showLatestPosts\(.*\);??\s*}\s*catch\(err\)\s*{\s*console\.log\(err\);?\s*}\s*}/);

Reviewer: I am unable to pass this test with the correct code
<img width="875" alt="Screenshot 2023-11-16 at 12 33 43 PM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/6c85c0da-c559-4357-ba19-64ab33c70d6e">

Author: done

  color: var(--black);
  text-decoration: none;
  display: block;
  width: fit-content;
  margin: 10px 0 10px;
}

.career {
  background-color: var(--salmon-color);
}

.feedback,
.html-css {
  background-color: var(--light-blue);
}

.support {
  background-color: var(--light-orange);
}

.general {
  background-color: var(--light-green);
}

.javascript {
  background-color: var(--golden-yellow);
}

.backend {
  background-color: var(--gold);
}

.python {
  background-color: var(--green);
}

.motivation {
  background-color: var(--peach);
}

.avatar-container {
  display: flex;
  justify-content: center;
  gap: 10px;
  flex-wrap: wrap;
}

.avatar-container img {
  width: 30px;
  height: 30px;
}

@media (max-width: 750px) {
  .table-wrapper {
    padding: 0 15px;
  }

  table {
    width: 700px;
  }

  th {
    font-size: 1.2rem;
  }

  .post-title {
    font-size: 1.1rem;
  }
}
```

```js
const FORUM_LATEST = "https://forum-proxy.freecodecamp.rocks/latest";
const FORUM_TOPIC_URL = "https://forum.freecodecamp.org/t/";
const FORUM_CATEGORY_URL = "https://forum.freecodecamp.org/c/";
const AVATAR_URL = "https://sea1.discourse-cdn.com/freecodecamp";

--fcc-editable-region--

> const fetchData = async () => { 
>   try{
>     const res = await fetch(FORUM_LATEST);
>     const data = await res.json();
>     console.log(data)
> 
>   }
>   catch(err){
>     console.log(err)
>   }
> }

Reviewer 2: ```suggestion
const fetchData = async () => { 
  try {
    const res = await fetch(FORUM_LATEST);
    const data = await res.json();
    console.log(data);
    
  } catch (err) {
    console.log(err);
  }
};
```

--fcc-editable-region--

fetchData();

> const showLatestPosts = (data) => {}
> 

Reviewer 2: ```suggestion
const showLatestPosts = (data) => {};
```
